### ISSUE TITLE
Dev: Allow domain managers to create child hosts

### ISSUE THREAD

**Issue by vickyszuchin** [cisagov/getgov#1741]
### Story

As a domain manager
I want to be able to add a host name/ip address combo that is not a nameserver for my domain
so that this host name can be used by ANOTHER .gov domain as a nameserver

### Acceptance Criteria

- [ ] Domain manager can add a host name/ip address combo that is not a nameserver for my domain
- [ ] Domain manager can add a domain that is a subdomain of another domain **without** providing IP addresses
- [ ] Domain manager can **not** create a nameserver that belongs to another .gov domain, it must already be created by that domain.

### Additional Context

Per Alysia:
- For example, example.gov wants to make the the host name ns1.example.gov with the ip 127.0.0.1. Then mydomain.gov is now able to use ns1.example.gov
- Allow nameservers added to a domain that is a subdomain of another domain without requiring IP addresses
- Don’t allow users to create a nameserver that belongs to another .gov domain, it must already be created by that domain.
- DB change may be required here

### Issue Links

Design ticket: #1740

**Comment by PaulKuykendall** [comment]
This will be revisited as part of DNS Hosting development effort.

**Comment by abroddrick** [comment]
@PaulKuykendall I think this one is more manage domains, not dns hosting. We dont need dns hosting at all to do this

**Comment by h-m-f-t** [comment]
@abroddrick, you're not wrong. The thinking though was that we'll be revisiting the entire DNS UI/UX then and it feels like then'd be a reasonable place to implement this.

**Comment by h-m-f-t** [comment]
This issue is really about creating _child hosts_. This would be a useful feature, but only domain managers of example.gov could create a child host for example.gov. We certainly wouldn't want others to do that. 

I'm going to change the title.
